### ISSUE TITLE
self:update still listed and attempts to run

### ISSUE THREAD

**Issue by gilzow** [platformsh/cli#18]
Currently on `Platform.sh CLI 4.0.0-rc.3` via brew

```
❯ brew list | grep platform
platformsh-cli
```
but `self:update` is still listed in the available commands (ie `platform list`) and will try to run the update if you attempt to run the command.

**Comment by lolautruche** [comment]
@gilzow Can you please upgrade to the latest version and check if it's still the case?

**Comment by gilzow** [comment]
```
❯ p --version
Platform.sh CLI 4.0.0-rc.3 (Wrapped legacy CLI 3.87.1)
❯ p self:update
Checking for Platform.sh CLI updates (current version: 3.87.1)

Version 4.0.0 is available. Release notes:
3.88.0:
  * Add support for automatic "self-installation" of shell configuration files.
  * Fix auto-login (broken in v3.87.0).
<snip>

Update to version 4.0.0? [Y/n] y
Updating to version 4.0.0
The Platform.sh CLI has been successfully updated to version 4.0.0
❯ p --version
Platform.sh CLI 4.0.0-rc.3 (Wrapped legacy CLI 3.87.1)
```

**Comment by lolautruche** [comment]
I meant upgrading using `brew` :)

**Comment by gilzow** [comment]
> I meant upgrading using brew :)

We both posted at the same time. The output above was from when I tested it to see what it would do. I want to include it for completeness. Upgrading via brew now... 

FYI, while updating via brew, I received this notice:
```
2023/01/06 18:14:16 PSH .phar file does not exist, copying: /private/tmp/psh-go-8.0.27-0.0.0/psh-0.0.0
Notice: date_default_timezone_set(): Timezone ID 'UTC0' is invalid in phar:///private/tmp/psh-go-8.0.27-0.0.0/psh-0.0.0/src/Application.php on line 50
Notice: date_default_timezone_set(): Timezone ID 'UTC0' is invalid in phar:///private/tmp/psh-go-8.0.27-0.0.0/psh-0.0.0/src/Application.php on line 50
```

```
❯ platform --version
Platform.sh CLI 4.0.0 (Wrapped legacy CLI 0.0.0)
```

`self:` commands are no longer listed in `platform list` and trying to do `self:update` results in
```
There are no commands defined in the "self" namespace.
```
as expected.

**Comment by lolautruche** [comment]
"Wrapped legacy CLI 0.0.0"
This is not expected, @akalipetis, is it?

**Comment by akalipetis** [comment]
@lolautruche opened #20 for that, closing this one as it was fixed with `4.0.0`.
